pallets/creditcoin/src/helpers/external_address.rs

```diff
@@ -1,14 +1,110 @@
 use crate::{Blockchain, ExternalAddress};
+use alloc::string::String;
 use base58::FromBase58;
+use core::convert::TryFrom;
+use frame_support::BoundedVec;
+use sp_core::ecdsa::Public;
+use sp_io::hashing::keccak_256;
 use sp_io::hashing::sha2_256;
+use sp_std::boxed::Box;
+use sp_std::vec::Vec;
+
+pub fn external_address_generator(
+	blockchain: &Blockchain,
+	reference: &ExternalAddress,
+) -> Option<Box<dyn Fn(Public) -> ExternalAddress>> {
+	match blockchain {
+		Blockchain::Luniverse | Blockchain::Ethereum | Blockchain::Rinkeby => {
+			match Etherlike::is_address(reference) {
+				Some(Etherlike::Simple) => Some(Box::new(Etherlike::from_public)),
+				Some(Etherlike::Eip55) => Some(Box::new(Etherlike::from_public_checksummed)),
+				_ => None,
+			}
+		},
+		Blockchain::Bitcoin => None,
+		Blockchain::Other(_) => None,
+	}
+}
+
+pub trait PublictoAddress {
+	type AddressType;
+	fn is_address(addr: &ExternalAddress) -> Option<Self::AddressType>;
+	fn from_public(pkey: Public) -> ExternalAddress;
+}
+
+pub enum Etherlike {
+	Simple,
+	Eip55,
```

Comment:
We're agnostic to Eip55, since we get the address's actual bytes instead of the hex-encoded address.

Reply:
Yes. What are you suggesting exactly? I'd like to keep the enum because that's how I plan to add support for all the bitcoin addresses for consistency. If someone ever wants/needs to use the checksum address it is already there even if there are ways to avoid it.

Reply:
I'm suggesting removing code around creating or handling checksummed ethereum addresses. On-chain logic will never receive or manipulate a checksummed address, so I don't see why we'd need this enum variant or the other code involving checksumming.